File: src/main/java/com/jobtracker/entity/JobApplication.java
package com.jobtracker.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "job_applications")
public class JobApplication {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotBlank(message = "Company name is required")
    @Column(nullable = false)
    private String companyName;
    
    @NotBlank(message = "Job title is required")
    @Column(nullable = false)
    private String jobTitle;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ApplicationStatus status;
    
    @Column(length = 2000)
    private String description;
    
    @Column(length = 1000)
    private String notes;
    
    private String location;
    
    private String jobUrl;
    
    private Integer salaryMin;
    
    private Integer salaryMax;
    
    private LocalDateTime appliedDate;
    
    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;
    
    // Constructors
    public JobApplication() {}
    
    // Getters and Setters
    public Long getId() {
        return id;
    }
    
    public void setId(Long id) {
        this.id = id;
    }
    
    public String getCompanyName() {
        return companyName;
    }
    
    public void setCompanyName(String companyName) {
        this.companyName = companyName;
    }
    
    public String getJobTitle() {
        return jobTitle;
    }
    
    public void setJobTitle(String jobTitle) {
        this.jobTitle = jobTitle;
    }
    
    public ApplicationStatus getStatus() {
        return status;
    }
    
    public void setStatus(ApplicationStatus status) {
        this.status = status;
    }
    
    public String getDescription() {
        return description;
    }
    
    public void setDescription(String description) {
        this.description = description;
    }
    
    public String getNotes() {
        return notes;
    }
    
    public void setNotes(String notes) {
        this.notes = notes;
    }
    
    public String getLocation() {
        return location;
    }
    
    public void setLocation(String location) {
        this.location = location;
    }
    
    public String getJobUrl() {
        return jobUrl;
    }
    
    public void setJobUrl(String jobUrl) {
        this.jobUrl = jobUrl;
    }
    
    public Integer getSalaryMin() {
        return salaryMin;
    }
    
    public void setSalaryMin(Integer salaryMin) {
        this.salaryMin = salaryMin;
    }
    
    public Integer getSalaryMax() {
        return salaryMax;
    }
    
    public void setSalaryMax(Integer salaryMax) {
        this.salaryMax = salaryMax;
    }
    
    public LocalDateTime getAppliedDate() {
        return appliedDate;
    }
    
    public void setAppliedDate(LocalDateTime appliedDate) {
        this.appliedDate = appliedDate;
    }
    
    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
    
    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
    
    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }
    
    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
    
    // Enum
    public enum ApplicationStatus {
        RESEARCHING,
        APPLIED,
        PHONE_SCREEN,
        TECHNICAL_INTERVIEW,
        ONSITE_INTERVIEW,
        OFFER,
        REJECTED,
        WITHDRAWN,
        ACCEPTED
    }
    
    // Business logic methods
    public boolean isActive() {
        return status != ApplicationStatus.REJECTED 
            && status != ApplicationStatus.WITHDRAWN 
            && status != ApplicationStatus.ACCEPTED;
    }
    
    public boolean isInInterviewProcess() {
        return status == ApplicationStatus.PHONE_SCREEN
            || status == ApplicationStatus.TECHNICAL_INTERVIEW
            || status == ApplicationStatus.ONSITE_INTERVIEW;
    }
}

File: frontend/src/types/JobApplication.ts
export enum ApplicationStatus {
  RESEARCHING = 'RESEARCHING',
  APPLIED = 'APPLIED',
  PHONE_SCREEN = 'PHONE_SCREEN',
  TECHNICAL_INTERVIEW = 'TECHNICAL_INTERVIEW',
  ONSITE_INTERVIEW = 'ONSITE_INTERVIEW',
  OFFER = 'OFFER',
  REJECTED = 'REJECTED',
  WITHDRAWN = 'WITHDRAWN',
  ACCEPTED = 'ACCEPTED',
}

export interface JobApplication {
  id: number;
  companyName: string;
  jobTitle: string;
  status: ApplicationStatus;
  description?: string;
  notes?: string;
  location?: string;
  jobUrl?: string;
  salaryMin?: number;
  salaryMax?: number;
  appliedDate?: string;
  createdAt: string;
  updatedAt: string;
  active: boolean;
  inInterviewProcess: boolean;
}

export interface JobApplicationRequest {
  companyName: string;
  jobTitle: string;
  status: ApplicationStatus;
  description?: string;
  notes?: string;
  location?: string;
  jobUrl?: string;
  salaryMin?: number;
  salaryMax?: number;
  appliedDate?: string;
}

export interface ApplicationStats {
  total: number;
  active: number;
  inInterview: number;
  rejected: number;
  accepted: number;
}

export const STATUS_LABELS: Record<ApplicationStatus, string> = {
  [ApplicationStatus.RESEARCHING]: 'Researching',
  [ApplicationStatus.APPLIED]: 'Applied',
  [ApplicationStatus.PHONE_SCREEN]: 'Phone Screen',
  [ApplicationStatus.TECHNICAL_INTERVIEW]: 'Technical Interview',
  [ApplicationStatus.ONSITE_INTERVIEW]: 'Onsite Interview',
  [ApplicationStatus.OFFER]: 'Offer',
  [ApplicationStatus.REJECTED]: 'Rejected',
  [ApplicationStatus.WITHDRAWN]: 'Withdrawn',
  [ApplicationStatus.ACCEPTED]: 'Accepted',
};

export const STATUS_COLORS: Record<ApplicationStatus, string> = {
  [ApplicationStatus.RESEARCHING]: 'default',
  [ApplicationStatus.APPLIED]: 'primary',
  [ApplicationStatus.PHONE_SCREEN]: 'secondary',
  [ApplicationStatus.TECHNICAL_INTERVIEW]: 'warning',
  [ApplicationStatus.ONSITE_INTERVIEW]: 'info',
  [ApplicationStatus.OFFER]: 'success',
  [ApplicationStatus.REJECTED]: 'error',
  [ApplicationStatus.WITHDRAWN]: 'default',
  [ApplicationStatus.ACCEPTED]: 'success',
};